### PR TITLE
Add Metric <-> Lightning Module integration tests

### PR DESCRIPTION
**Pull request by teddykoker**
Test Lightning integration with metrics

### CODE REVIEW

File: tests/metrics/test_metric_lightning.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+        def training_step(self, batch, batch_idx):
+            x = batch
+            self.metric(x.sum())
+            self.sum += x.sum()
+
+            return self.step(x)
+
+        def training_epoch_end(self, outs):
+            assert torch.allclose(self.sum, self.metric.compute())
```
Comment (Borda) [inline tests/metrics/test_metric_lightning.py:34]: now much close, lets spec tolerance, also why not exact?

Comment (teddykoker) [reply tests/metrics/test_metric_lightning.py:34]: default tolerance is 1e-8, i can parameterize if you want. Can't do exact bc of floating point

File: tests/metrics/test_metric_lightning.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+def test_metric_lightning(tmpdir):
+    class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.metric = SumMetric()
+            self.sum = 0.0
+
+        def training_step(self, batch, batch_idx):
```
Comment (ananyahjha93) [inline tests/metrics/test_metric_lightning.py:26]: shouldn't self.metric() be taking in batch_idx?

Comment (teddykoker) [reply tests/metrics/test_metric_lightning.py:26]: why?

Comment (teddykoker) [reply tests/metrics/test_metric_lightning.py:26]: I'm just summing all the values from the batch

**Pull request merged**